grails-app/services/org/pih/warehouse/core/IdentifierService.groovy

```diff
@@ -166,7 +166,41 @@ class IdentifierService {
         else if (identifier.length() > maxSize) {
             identifier = identifier.substring(0, maxSize)
         }
-        return identifier.toUpperCase()
+
+        // Checking for a duplicates
+        if (validateOrganizationIdentifier(identifier.toUpperCase())) {
+            return identifier.toUpperCase()
+        }
+
+        if (Character.isDigit(identifier.charAt(identifier.size() - 1))) {
+            String identifierWithHighestNumber = getOrganizationIdentifierWithHighestSuffix(identifier)
+            char suffix = identifierWithHighestNumber.charAt(identifierWithHighestNumber.size() - 1)
+            suffix++
+
+            return identifier.toUpperCase().substring(0, identifier.size() -1) + suffix
+        } else {
+            String identifierWithHighestNumber = getOrganizationIdentifierWithHighestSuffix(identifier)
```

Comment:
you are doing the substring in the getOrganizationIdentifierWithHighestSuffix even if there is no digit at the end

Reply:
yes, it is because there is possibility to create identifier without number, and there still can be another one in db, so I would like to check it and increase the highest number.